bundles/org.openhab.binding.hdpowerview/src/test/java/org/openhab/binding/hdpowerview/ShadePositionTest.java

```diff
@@ -403,4 +403,58 @@ public void testType44ShadePositionParsingShadeFullyDownVaneOpen() {
         assertShadePosition(test.getState(capabilities, SECONDARY_POSITION), UnDefType.UNDEF);
         assertShadePosition(test.getState(capabilities, VANE_TILT_POSITION), 88);
     }
+
+    /**
+     * Test the getCapabilities functionality.
+     */
+    @Test
+    public void testGetCapabilities() {
+        Capabilities caps;
+        /*
+         * - type not in database
+         * - null external capabilities
+         * => return invalid (-1)
+         */
+        caps = db.getCapabilities(0, null);
+        assertEquals(-1, caps.getValue());
+        /*
+         * - type not in database
+         * - valid external capabilities (1)
+         * => return external capabilities (1)
+         */
+        caps = db.getCapabilities(0, 1);
+        assertEquals(1, caps.getValue());
+        /*
+         * - type not in database
+         * - invalid external capabilities (99)
+         * => return invalid (-1)
+         */
+        caps = db.getCapabilities(0, 99);
+        assertEquals(-1, caps.getValue());
+        /*
+         * - type 1 in database
+         * - inherent capabilities (0)
+         * - null external capabilities
+         * => return inherent capabilities (0)
+         */
+        caps = db.getCapabilities(1, null);
+        assertEquals(0, caps.getValue());
+        /*
+         * - type 1 in database
+         * - inherent capabilities (0)
+         * - non matching external capabilities (1)
+         * => return inherent capabilities (0)
+         */
+        caps = db.getCapabilities(1, 1);
+        assertEquals(0, caps.getValue());
```

Comment:
Please see other comment. Here I would have expected 1 as directly provided by hub, i.e. taking precedence over the mismatching capabilities from type (0).

Reply:
See sbove.